source/MaterialXGenMdl/MdlShaderGenerator.cpp

```diff
@@ -343,7 +327,16 @@ ShaderPtr MdlShaderGenerator::generate(const string& name, ElementPtr element, G
 string MdlShaderGenerator::getUpstreamResult(const ShaderInput* input, GenContext& context) const
 {
     const ShaderOutput* upstreamOutput = input->getConnection();
-    if (!upstreamOutput || upstreamOutput->getNode()->isAGraph())
+
+	// TODO: This is a temporary fix for Iray.
+	// File texture constructors with a filename set are emitted inline "by value".
+	if (upstreamOutput && upstreamOutput->getType() == Type::FILENAME &&
+		upstreamOutput->getValue() && !upstreamOutput->getValue()->getValueString().empty())
+	{
+		return _syntax->getValue(upstreamOutput->getType(), *upstreamOutput->getValue());
```

Comment:
Need to emit file textures (texture_2d object) inline instead of first assigning to a variable, since texture_2d is not allowed as a variable in all contexts.